packages/office-ui-fabric-react/src/components/Fabric/Fabric.base.tsx

```diff
@@ -35,17 +35,23 @@ export class FabricBase extends React.Component<
 
   public render() {
     const { as: Root = 'div', theme, dir } = this.props;
-    const classNames = this._getClassNames();
-    const divProps = getNativeProps<React.HTMLAttributes<HTMLDivElement>>(this.props, divProperties);
     const isRTL = dir === 'rtl';
-    let renderedContent = <Root {...divProps} className={classNames.root} ref={this._rootElement} />;
 
-    // Expose an rtl based theme if dir is specified and it doesn't agree with the theme setting.
+    // Adjust dir if prop is specified and it doesn't agree with the theme setting.
     if (dir !== undefined && theme && (theme.rtl === undefined || theme.rtl !== isRTL)) {
-      renderedContent = <Customizer settings={{ theme: getRTLTheme(theme, isRTL) }}>{renderedContent}</Customizer>;
+      const newTheme = getRTLTheme(theme, isRTL);
```

Comment:
There is some duplication in the `if` and `else` branches, but I think pulling out the common parts and making completely DRY would lead to a brittle result.

Reply:
@JasonGore and I talked about this idea when I added RTL to Fabric. Adding customizer to Fabric might be overloading the control a bit. 

Should we be creating a new control that provides theme and Fabric together? For instance, our layer uses `Fabric`, so we're loading customizer with just about every control now.

Reply:
Oh boy, I didn't realize (or remember) when we talked that `Layer` was actually using `Fabric` as well. I wonder what the rationale is for `Layer` to use `Fabric`? If it's simply to provide context, I don't think that's needed any more with React portals. If it's to watch for keypresses and use focus outlining, I think that would also propagate through React portals now, as well. We should explore removing `Fabric` from `Layer` entirely.

In any case, it seems not good to have `Fabric` and `Customizer` for every layered component (ContextualMenu, Callout, Dropdown, DatePicker, DetailsHeader, KeytipLayer, Dialog, Modal, Panel, etc.)